pkg/app/api/grpcapi/piped_api.go

```diff
@@ -82,6 +87,7 @@ func NewPipedAPI(ctx context.Context, ds datastore.DataStore, sls stagelogstore.
 		deploymentPipedCache:      memorycache.NewTTLCache(ctx, 24*time.Hour, 3*time.Hour),
 		envProjectCache:           memorycache.NewTTLCache(ctx, 24*time.Hour, 3*time.Hour),
 		pipedStatCache:            hc,
+		redis:                     rd,
```

Comment:
should we use the `redis` explicitly here or just pass `pipedCfgCache` as we do for `pipedStatCache`? 🤔

Reply:
I feel you. Honestly I also want to abstract it in such a way. But we want to generate HashCache with different keys for each project in the method, that's why PipedAPI needs to have a connection to the redis server.

https://github.com/pipe-cd/pipe/pull/2847/files#diff-dc8d28c3b055fcad0f022406b744ce756e3be8c630b6c40aef12e026c6a58dcfR1014

Reply:
ah okay I see, get the point here, thanks 👍

Reply:
Just created this issue for those things, lets think about it later 👍

Reply:
Nice 👍